File: Array/linchuanxin/Algorithm/day05_0605_链表相交/day05.java
 /**
     * Definition for singly-linked list.
     * public class ListNode {
     *     int val;
     *     ListNode next;
     *     ListNode(int x) {
     *         val = x;
     *         next = null;
     *     }
     * }
     */
    public class Solution {
        public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
            if(headA == null || headB == null)
                return null;
            ListNode p = headA;
            ListNode q = headB;
            while(p != q){
                if(p == null){
                    p = headB;
                }else{
                    p = p.next;
                }
                if(q == null){
                    q = headA;
                }else{
                    q = q.next;
                }
            }

            return p;
        }
    }

File: Array/linchuanxin/Algorithm/day05_0605_链表相交/day05.c
#include <stdio.h>

struct ListNode {
    int val;
    struct ListNode *next;
};

struct ListNode *getIntersectionNode(struct ListNode *headA, struct ListNode *headB){
    if(headA == NULL||headB == NULL)
        return NULL;
    struct ListNode *p = headA;
    struct ListNode *q = headB;

    while (q != p){
        if (p == NULL) {
            p = headB;
        } else {
            p = p->next;
        }
        if (q == NULL) {
            q = headB;
        } else {
            q = q->next;
        }
    }
    return p;
}